File: src/main/java/net/glasslauncher/mods/glassguis/screen/widget/slot/CustomSizeSlot.java
package net.glasslauncher.mods.glassguis.screen.widget.slot;

import net.modificationstation.stationapi.api.client.texture.atlas.Atlas;
import org.jetbrains.annotations.Nullable;

public interface CustomSizeSlot {
    int getWidth();
    int getHeight();

    default boolean keepBackgroundTexture() {
        return false;
    }

    default int getBackgroundWidth() {
        return getWidth();
    }

    default int getBackgroundHeight() {
        return getHeight();
    }

    default boolean shouldScaleItem() {
        return false;
    }

    @Nullable
    default Atlas.Sprite getBackgroundSprite() {
        return null;
    }
}

File: src/main/kotlin/net/glasslauncher/mods/glassguis/CustomSizeSlot.kt
package net.glasslauncher.mods.glassguis

import net.modificationstation.stationapi.api.client.texture.Sprite
import net.modificationstation.stationapi.api.client.texture.atlas.Atlas

interface CustomSizeSlot {
    fun getWidth(): Int
    fun getHeight(): Int

    fun keepBackgroundTexture(): Boolean {
        return false
    }

    fun getBackgroundWidth(): Int {
        return getWidth()
    }

    fun getBackgroundHeight(): Int {
        return getHeight()
    }

    fun shouldScaleItem(): Boolean {
        return false
    }

    fun getBackgroundSprite(): Atlas.Sprite? {
        return null
    }
}